fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryTableFunctionPlan.java

```diff
@@ -0,0 +1,280 @@
+        if (this == otherExpression) {
+            return true;
+        }
+        if (getClass() != otherExpression.getClass()) {
+            return false;
+        }
+        final var otherTableFunctionPlan =  (RecordQueryTableFunctionPlan)otherExpression;
+
+        return value.semanticEquals(otherTableFunctionPlan.value, equivalencesMap) &&
+                semanticEqualsForResults(otherExpression, equivalencesMap);
+    }
+
+    @SuppressWarnings("EqualsWhichDoesntCheckParameterClass")
+    @Override
+    public boolean equals(final Object other) {
+        return structuralEquals(other);
+    }
+
+    @Override
+    public int hashCode() {
+        return structuralHashCode();
+    }
+
+    @Override
+    public int hashCodeWithoutChildren() {
+        return Objects.hash(getResultValue());
+    }
+
+    @Override
+    public void logPlanStructure(StoreTimer timer) {
+        // nothing to increment
+    }
+
+    @Override
+    public int getComplexity() {
+        return 1;
+    }
+
+    @Override
+    public int planHash(@Nonnull final PlanHashMode mode) {
+        switch (mode.getKind()) {
+            case LEGACY:
+            case FOR_CONTINUATION:
+                return PlanHashable.objectsPlanHash(mode, BASE_HASH, getResultValue());
+            default:
+                throw new UnsupportedOperationException("Hash kind " + mode.getKind() + " is not supported");
+        }
+    }
+
+    @Nonnull
+    @Override
+    public PlannerGraph rewritePlannerGraph(@Nonnull final List<? extends PlannerGraph> childGraphs) {
+        return PlannerGraph.fromNodeAndChildGraphs(
+                new PlannerGraph.OperatorNodeWithInfo(this,
+                        NodeInfo.VALUE_COMPUTATION_OPERATOR,
+                        ImmutableList.of("TFUNC {{expr}}"),
+                        ImmutableMap.of("expr", Attribute.gml(value.toString()))),
+                childGraphs);
+    }
+
+    @Nonnull
+    @Override
+    public PRecordQueryTableFunctionPlan toProto(@Nonnull final PlanSerializationContext serializationContext) {
+        return PRecordQueryTableFunctionPlan.newBuilder()
+                .setValue(value.toValueProto(serializationContext))
+                .build();
+    }
+
+    @Nonnull
+    @Override
+    public PRecordQueryPlan toRecordQueryPlanProto(@Nonnull final PlanSerializationContext serializationContext) {
+        return PRecordQueryPlan.newBuilder().setTableFunctionPlan(toProto(serializationContext)).build();
+    }
+
+    @Nonnull
+    public static RecordQueryTableFunctionPlan fromProto(@Nonnull final PlanSerializationContext serializationContext,
+                                                         @Nonnull final PRecordQueryTableFunctionPlan recordQueryTableFunctionPlanProto) {
+        final var value = Value.fromValueProto(serializationContext,
+                Objects.requireNonNull(recordQueryTableFunctionPlanProto.getValue()));
+        if (!(value instanceof StreamingValue)) {
```

Comment:
This is guaranteed by the class definition to be a `StreamingValue`. In the very remote case that we change that without afterthought I think a lot of things would break anyway so why not just cast it?

Reply:
It is extra defensive, considering the `proto` message is something sent from an external client.